gossip/src/duplicate_shred_handler.rs

```diff
@@ -144,18 +176,63 @@ impl DuplicateShredHandler {
             Some(SlotStatus::Frozen) => {
                 return false;
             }
-            Some(SlotStatus::UnfinishedProof(slot_map)) => {
-                if let Some(proof_chunkmap) = slot_map.peek(&data.from) {
-                    if proof_chunkmap.wallclock < data.wallclock {
-                        return false;
+            Some(SlotStatus::UnfinishedProof(slot_map)) => match slot_map.get(&data.from) {
+                None => {
+                    if slot_map.len() < MAX_PUBKEY_PER_SLOT {
```

Comment:
can this `if` check go inside `check_has_enough_stake_and_cleanup`.

Reply:
done